db/change_listener.go

```diff
@@ -69,9 +69,13 @@ func (listener *changeListener) StartMutationFeed(bucket base.Bucket, dbStats *e
 			return err
 		}
 		go func() {
-			defer close(listener.FeedArgs.DoneChan)
-			defer base.FatalPanicHandler()
-			defer listener.notifyStopping()
+			defer func() {
+				if listener.FeedArgs.DoneChan != nil {
+					close(listener.FeedArgs.DoneChan)
+				}
+				base.FatalPanicHandler()
+				listener.notifyStopping()
+			}()
```

Comment:
As discussed, restore order of defer processing.

Reply:
Sure, included this change in the latest commit.